lib/pure/net.nim

```diff
@@ -120,21 +120,16 @@ when defineSsl:
     SslContext* = ref object
       context*: SslCtx
       referencedData: HashSet[int]
-      extraInternal: SslContextExtraInternal
 
     SslAcceptResult* = enum
       AcceptNoClient = 0, AcceptNoHandshake, AcceptSuccess
 
     SslHandshakeType* = enum
       handshakeAsClient, handshakeAsServer
 
-    SslClientGetPskFunc* = proc(hint: string): tuple[identity: string, psk: string]
+    SslClientGetPskFunc* = proc(hint: string): tuple[identity: string, psk: string]{.nimcall.}
```

Comment:
technically, anything other than {.closure.} is fine, is this over-restrictive?

Reply:
I don't see how it is over-restrictive, the last thing we need is more generic bloat in Nim's core libraries.